scope.go

```diff
@@ -183,10 +183,15 @@ func (s *scope) incQueued() error {
 		// Choose new host, since the previous one may become obsolete
 		// after sleeping.
 		h := s.cluster.getHost()
+		// if request has session_id, set same host
+		if s.sessionId != "" {
+			h = s.cluster.getHostSticky(s.sessionId)
+		}
```

Comment:
`s.cluster.getHost()` modifies state with
```
idx := atomic.AddUint32(&c.nextReplicaIdx, 1)
```
therefore you'd rather do:
```suggestion
		// if request has session_id, set same host
		if s.sessionId != "" {
			h = s.cluster.getHostSticky(s.sessionId)
		} else {
		        h := s.cluster.getHost()
		}
```

Reply:
I agree